bin/yb-ctl

```diff
@@ -694,10 +756,48 @@ class ClusterOptions:
         raise RuntimeError("No binary found for {}. Considered binary directories: {}".format(
             binary_name, binary_dirs))
 
-    def get_address(self, daemon_id, port_type, include_base_url=False):
-        port_str = str(self.base_ports[daemon_id.daemon_type][port_type])
+    def get_host_port(self, daemon_id, port_type):
```

Comment:
we can use the same behavior here as well, instead of ysql_protocol, the calling functions could explicitly pass in the keys in the `self.base_ports`